homeworks/homework02/homework02.Rmd

```diff
@@ -0,0 +1,166 @@
+**10 points**
+
+Select the following columns from the `data` variable you created above. 
+`ensembl_gene_id`, 
+`CD34_shRPL5_RPF_1`, `CD34_shRPL5_RPF_2`, `CD34_shRPL5_RNA_1`, `CD34_shRPL5_RNA_2`,
+`CD34_shRPS19_RPF_1`, `CD34_shRPS19_RPF_2`, `CD34_shRPS19_RNA_1`, `CD34_shRPS19_RNA_2`,
+`CD34_shLuc_RPF_1`, `CD34_shLuc_RPF_2`, `CD34_shLuc_RNA_1`, `CD34_shLuc_RNA_2`.
+
+## Problem 5
+
+**10 points**
+
+Filter the result from Problem 4 to include only rows where each of the 12 numerical columns 
+you selected above has 50 counts or more. This is a simple way to avoid genes that have
+very low counts. You might be tempted to do this step separately for
+each of the 12 columns. But you can write lot less code if you use the `filter_if`
+and `all_vars` functions you learned above.
+
+## Problem 6
+
+**10 points**
+
+After filtering in Problem 5, divide each of the above 12 numerical columns by the 
+corresponding median value in each column. This median normalization is typically done in 
+high-throughput experiments to normalize for sample-to-sample difference in read depth.
+Again, you can write lot less code if you use the `mutate_if` and `funs` functions 
+you learned above.
+
+## Problem 7
+
+**10 points**
+
+After we do the above filtering and median-normalization, let us calculate 
+translation efficiency as the average ratio of the RPF and RNA reads for each
+treatment condition. Then we calculate how this translation efficiency changes
+between target (`rpl5` and `rps19`) and control (`luc`) shRNAs.
+
+The code implementing the above steps is shown below, but it has a few
+errors. Correct them.
+
+```{r}
+lfc <- data %>% 
+  mutate(mean_rpl5_te = ((CD34_shRPL5_RPF_1 + CD34_shRPL5_RPF_2) / 
+                            (CD34_shRPL5_RNA_1 + CD34_shRPL5_RNA_2)) %>% 
+  mutate(mean_rps19_te = ((CD34_shRPS19_RPF_1 + CD34_shRPS19_RPF_2) / 
+                            (CD34_shRPS19_RNA_1 + CD34_shRPS19_RNA_2)) %>% 
+  mutate(mean_shluc_te = ((CD34_shLuc_RPF_1 + CD34_shLuc_RPF_2) / 
+                            (CD34_shLuc_RNA_1 + CD34_shLuc_RNA_2)) %>% 
+  select(ensembl_gene_id, mean_rpl5_te, mean_rps19_te) %>% 
+  mutate(lfc_te_rpl5 == log2(mean_rpl5_te / mean_shluc_te),
+         lfc_te_rps19 == log2(mean_rps19_te / mean_shluc_te)) %>% 
+  print()
+```
+
+## Problem 8
+
+**10 points**
+
+Create a new dataframe called `mean_lfc` from `lfc` 
+containing a new column called `avg_lfc`.
+`avg_lfc` should be the average of the log2 fold-change in TE (`lfc_te`) upon
+knockdown of RPL5 and RPS19.
+
+Then select only the gene id column and the new column that you just created.
+
+## Problem 9
+
+**10 points**
+
+Join the above `mean_lfc` dataframe with the `annotations` dataframe created
+at the top of the document.
+
+
+## Problem 10
+
+**5 points**
+
+Select only the bottom 10 genes with the lowest `avg_lfc` and display the
+gene `symbol`, gene `name` and `avg_lfc` for these genes.
+
+5 bonus points if you use one of the new functions you learned above!
```

Comment:
Are the new functions referenced here from Problem 1?

Reply:
@k8hertweck Yes, I have clarified in this updated commit 5c983b2